src/db/repository/UserRepository.ts

```diff
@@ -23,6 +23,18 @@ export default class UserRepository {
 		}
 	}
 
+	async getUserAndUpdateById(id: string, updateParams: Partial<User>) {
+		try {
+			const updatedUser = await this.userModel
+				.findOneAndUpdate({ _id: id }, { $set: updateParams }, { new: true })
+				.lean()
+				.exec();
+			return updatedUser;
```

Comment:
```suggestion
			return await this.userModel
				.findOneAndUpdate({ _id: id }, { $set: updateParams }, { new: true })
				.lean()
				.exec();
```

@nguyenit67 here's a suggestion you can just apply to match @Enubia's change request so you don't have to do it manually.

Reply:
Thanks for the change suggestion, couldn't figure out how to do that in the mobile app haha

Reply:
@nick-w-nick thanks pal!